grr/server/grr_response_server/gui/ui/components/client/client_test.ts

```diff
@@ -36,59 +45,121 @@ describe('Client Component', () => {
             ApiModule,
             NoopAnimationsModule,
             ClientModule,
+            ClientDetailsModule,
+            RouterTestingModule.withRoutes(CLIENT_ROUTES),
           ],
           providers: [
             {
               provide: ActivatedRoute,
               useValue: {
                 paramMap: paramsSubject,
+                snapshot: {},
               },
             },
             {provide: ConfigFacade, useFactory: () => configFacade},
             {provide: UserFacade, useFactory: () => userFacade},
+            {provide: Router, useValue: {}},
           ],
 
         })
         .compileComponents();
 
-    facade = TestBed.inject(ClientPageFacade);
+    clientPageFacade = TestBed.inject(ClientPageFacade);
+    clientDetailsFacade = TestBed.inject(ClientDetailsFacade);
+    location = TestBed.get(Location);
+    router = TestBed.get(Router);
   }));
 
   it('loads client information on route change', () => {
     const fixture = TestBed.createComponent(ClientComponent);
     fixture.detectChanges();  // Ensure ngOnInit hook completes.
 
-    const searchClientsSpy = spyOn(facade, 'selectClient');
+    const searchClientsSpy = spyOn(clientPageFacade, 'selectClient');
     paramsSubject.next(new Map(Object.entries({id: 'C.1234'})));
     fixture.detectChanges();
 
     expect(searchClientsSpy).toHaveBeenCalledWith('C.1234');
   });
 
-  it('displays client details on client change', () => {
-    // Prevent warnings from 404-ing API requests.
-    spyOn(facade, 'selectClient');
-
-    const subject = new Subject<Client>();
-    Object.defineProperty(facade, 'selectedClient$', {get: () => subject});
-
-    const fixture = TestBed.createComponent(ClientComponent);
-    fixture.detectChanges();  // Ensure ngOnInit hook completes.
-
-    paramsSubject.next(new Map(Object.entries({id: 'C.1234'})));
-    subject.next({
-      clientId: 'C.1234',
-      fleetspeakEnabled: true,
-      knowledgeBase: {
-        fqdn: 'foo.unknown',
-      },
-      lastSeenAt: new Date(1571789996678),
-      labels: [],
-    });
-    fixture.detectChanges();
-
-    const text = fixture.debugElement.nativeElement.textContent;
-    expect(text).toContain('C.1234');
-    expect(text).toContain('foo.unknown');
-  });
+  it('correctly updates URL when navigating from main page to details page',
+     fakeAsync(() => {
+       // Prevent warnings from 404-ing API requests.
+       spyOn(clientPageFacade, 'selectClient');
+       spyOn(clientDetailsFacade, 'selectClient');
+
+       const subject = new Subject<Client>();
+       Object.defineProperty(clientPageFacade, 'selectedClient$', {get: () => subject});
+       spyOn(clientPageFacade, 'removeClientLabel');
+
+       router.navigate(['clients/C.1234']);
```

Comment:
This should fail because `router` is injected to be `{}`. Can you validate that this works?

Reply:
I updated the tests